pkg/openstack/loadbalancer_test.go

```diff
@@ -973,3 +974,89 @@ func TestLbaasV2_updateServiceAnnotations(t *testing.T) {
 
 	assert.ElementsMatch(t, expectedAnnotations, serviceAnnotations)
 }
+
+func TestLbaasV2_getMemberSubnetID(t *testing.T) {
+	lbaasOpts := LoadBalancerOpts{
+		LBClasses:      map[string]*LBClass{"lbclassKey": {MemberSubnetID: "lb-class-member-subnet-id-5678"}},
+		MemberSubnetID: "lb-class-memberSubnetId",
+	}
+
+	tests := []struct {
+		name    string
+		opts    LoadBalancerOpts
+		service *corev1.Service
+		svcConf *serviceConfig
+		want    string
+		wantErr error
```

Comment:
i can't check a nil error if it was string

Reply:
@pierreprinetti I just made some changes you suggested and made some of the modification in a PR here 
https://github.com/kubernetes/cloud-provider-openstack/pull/2438

Reply:
> i can't check a nil error if it was string the program returns nil sometimes, hence the error check

If `wantErr` is a string, you can say that when it's the empty string, then we don't expect an error. This works well because `assert.EqualError` accepts a string as the expected error.

if wantErr != "" { assert.EqualError(...) } else { assert.NoError(...) }

Reply:
yeah you are perfectly right, thank you
i should be checking the state of what the test wanted instead of the return value; that was my mistake ;)

Reply:
@pierreprinetti  i took all your advice and made the neccesary changes already
can you re-review please